stdlib/public/SDK/Foundation/Data.swift

```diff
@@ -1206,10 +1196,8 @@ public struct Data : ReferenceConvertible, Equatable, Hashable, RandomAccessColl
             try $0.write(to: url, options: WritingOptions(rawValue: options.rawValue))
 #else
             if _shouldUseNonAtomicWriteReimplementation(options: options) {
-                var error : NSError?
-                if !_NSWriteDataToFile_Swift(url: url._bridgeToObjectiveC(), data: $0, options: options.rawValue, error: &error) {
-                    throw error!
-                }
+                var error: NSError? = nil
+                guard __NSDataWriteToURL($0, url, options, &error) else { throw error! }
```

Comment:
Strange. For now, can you try adding `__attribute__((swift_error(none)))` (`NS_SWIFT_NOTHROW`) to the function declaration on the C side? Maybe some part of the code is still trying to make this a throwing function.